### PR TITLE
Add logic for immutable collection converters to share create-object delegate

### PR DESCRIPTION
**Pull request by layomia**
… rather than have each converter create one.

### CODE REVIEW

File: src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ImmutableDictionaryOfStringTValueConverter.cs [hunk 1]
```diff
@@ -28,7 +28,15 @@ protected override void CreateCollection(ref ReadStack state)
 
         protected override void ConvertCollection(ref ReadStack state, JsonSerializerOptions options)
         {
-            state.Current.ReturnValue = GetCreatorDelegate(options)((Dictionary<string, TValue>)state.Current.ReturnValue!);
+            JsonClassInfo classInfo = state.Current.JsonClassInfo;
+
+            if (classInfo.CreateObjectWithArgs == null)
+            {
+                classInfo.CreateObjectWithArgs = options.MemberAccessorStrategy.CreateImmutableDictionaryCreateRangeDelegate<TValue, TCollection>();
```
Comment (steveharter) [inline src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ImmutableDictionaryOfStringTValueConverter.cs:35]: nit: add temp variable so `classInfo.CreateObjectWithArgs` doesn't need to be called below.

**Review by steveharter — APPROVED** [review]
LGTM; one nit on perhaps using a temp variable (applies to more than one location)

**Review by jozkee — APPROVED** [review]
Is this the only set of converters where we can apply this pattern?

**Comment by layomia** [comment]
> Is this the only set of converters where we can apply this pattern?

[`IEnumerableWithAddMethodConverter<T>`](https://github.com/dotnet/runtime/blob/b4439ca1fd994c8ec2592fb9614a841853ae08fb/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IEnumerableWithAddMethodConverter.cs#L10) can benefit from this as well. I presumed while writing this PR that the delegate here needs to live on `ReadStackFrame`, but I see now that it doesn't. I'll open another PR, thanks.

**Comment by layomia** [comment]
Test failures unrelated.

**Pull request merged**